tools/runfiles/runfiles.rs

```diff
@@ -100,10 +130,39 @@ impl Runfiles {
         if path.is_absolute() {
             return path.to_path_buf();
         }
+
+        let path = self
+            .repo_mapping
+            .as_ref()
+            .and_then(|mapping| {
+                let mut components = path.components();
+                if let Some(std::path::Component::Normal(target_local)) = components.next() {
+                    if let Some(target_local) = target_local.to_str() {
+                        let current_canonical = self.current_repository();
```

Comment:
We're looking at this PR with @Wyverald, and all seems fine apart from this line; we concluded that the implementation of `current_repository()` is currently incorrect. We'll first need to tackle that, because it causes the next couple of lines to be wrong as well. I'll file a separate issue to describe the problem.

Reply:
@scentini Has `current_repository` been fixed? If not, could you link to the bug that was filed so I can know when it is fixed.